### ISSUE TITLE
Internal Assertion error

### ISSUE THREAD

**Issue by Arunavameister** [dbolya/yolact#275]
Hello,

I am trying to train a custom dataset(with just a handful images for now) to try out this tool on GOOGLE COLAB, however i am getting the following error:
!python train.py --config=yolact_base_config --batch_size=1
___________________________________________________________________

Scaling parameters by 0.12 to account for a batch size of 1.
Per-GPU batch size is less than the recommended limit for batch norm. Disabling batch norm.
loading annotations into memory...
Done (t=0.00s)
creating index...
index created!
loading annotations into memory...
Done (t=0.00s)
creating index...
index created!
Initializing weights...
Begin training!

Traceback (most recent call last):
  File "train.py", line 502, in <module>
    train()
  File "train.py", line 305, in train
    losses = net(datum)
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py", line 541, in __call__
    result = self.forward(*input, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/parallel/data_parallel.py", line 150, in forward
    return self.module(*inputs[0], **kwargs[0])
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py", line 541, in __call__
    result = self.forward(*input, **kwargs)
  File "train.py", line 144, in forward
    losses = self.criterion(self.net, preds, targets, masks, num_crowds)
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py", line 541, in __call__
    result = self.forward(*input, **kwargs)
  File "/content/drive/My Drive/YOLACT/layers/modules/multibox_loss.py", line 124, in forward
    loc_t, conf_t, idx_t, idx, loc_data[idx])
  File "/content/drive/My Drive/YOLACT/layers/box_utils.py", line 182, in match
    overlaps = jaccard(truths, decoded_priors) if not cfg.use_change_matching else change(truths, decoded_priors)
  File "/content/drive/My Drive/YOLACT/layers/box_utils.py", line 73, in jaccard
    inter = intersect(box_a, box_b)
RuntimeError: strides[cur - 1] == sizes[cur] * strides[cur] INTERNAL ASSERT FAILED at /pytorch/torch/csrc/jit/fuser/executor.cpp:175, please report a bug to PyTorch. 
The above operation failed in interpreter, with the following stack trace:

_____________________________________________________________________

Any help will be appreciated!
Thanks

**Comment by dzyjjpy** [comment]
Met similar issue. Do you solve the issue? @Arunavameister 

Here is my issue:
Traceback (most recent call last):
  File "/home/adminroot/pycharm-community-2019.1.3/helpers/pydev/pydevd.py", line 1758, in <module>
    main()
  File "/home/adminroot/pycharm-community-2019.1.3/helpers/pydev/pydevd.py", line 1752, in main
    globals = debugger.run(setup['file'], None, None, is_module)
  File "/home/adminroot/pycharm-community-2019.1.3/helpers/pydev/pydevd.py", line 1147, in run
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/home/adminroot/pycharm-community-2019.1.3/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/home/jiapy/workspace/yolact/train.py", line 506, in <module>
    train()
  File "/home/jiapy/workspace/yolact/train.py", line 309, in train
    losses = net(datum)
  File "/home/jiapy/virtualEnv/py3.6torch1.0.1/lib/python3.6/site-packages/torch/nn/modules/module.py", line 489, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/jiapy/virtualEnv/py3.6torch1.0.1/lib/python3.6/site-packages/torch/nn/parallel/data_parallel.py", line 139, in forward
    inputs, kwargs = self.scatter(inputs, kwargs, self.device_ids)
  File "/home/jiapy/workspace/yolact/train.py", line 161, in scatter
    splits = prepare_data(inputs[0], devices, allocation=args.batch_alloc)
  File "/home/jiapy/workspace/yolact/train.py", line 432, in prepare_data
    split_images[device_idx]    = torch.stack(images[cur_idx:cur_idx+alloc], dim=0)
RuntimeError: expected a non-empty list of Tensors

**Comment by Arunavameister** [comment]
Unfortunately no

**Comment by Akash-Bhiwgade** [comment]
Hi,

I am getting the same error while training the model on custom dataset. I have got the dataset formatted as per COCO requirements.

`# print(box_b[:, :, 2:].unsqueeze(1).expand(n, A, B, 2))    [ Variable[CUDAFloatType]{1,0,19248,2} ]
    a = box_b[:, :, 2:].unsqueeze(1).expand(n, A, B, 2)
    # print(box_b[:, :, :2].unsqueeze(1).expand(n, A, B, 2))    [ Variable[CUDAFloatType]{1,0,19248,2} ]
    b = box_b[:, :, :2].unsqueeze(1).expand(n, A, B, 2)
    # print(box_a[:, :, 2:].unsqueeze(2).expand(n, A, B, 2))    [ Variable[CUDAFloatType]{1,0,19248,2} ]
    c = box_a[:, :, 2:].unsqueeze(2).expand(n, A, B, 2)
    # print(box_a[:, :, :2].unsqueeze(2).expand(n, A, B, 2))    [ Variable[CUDAFloatType]{1,0,19248,2} ]
    d = box_a[:, :, :2].unsqueeze(2).expand(n, A, B, 2)`

I have tried to debug the code in the box_utils.py file where the code actually fails to proceed ahead. The issue I see is when the code tries to do torch.clamp like below:
`inter = torch.clamp((max_xy - min_xy), min=0)`

Any reference or hints on fixing this error would be really appreciated.

Attached are the training and GPU stats for reference.

![Screenshot from 2020-01-16 16-36-28](https://user-images.githubusercontent.com/32064265/72520076-b01dd000-387e-11ea-9fa6-e5f4f4efa529.png)
![Screenshot from 2020-01-16 16-36-56](https://user-images.githubusercontent.com/32064265/72520077-b0b66680-387e-11ea-8b98-2e79e2f14e1f.png)

Best Regards,
Akash Bhiwgade

**Comment by aakashjhawar** [comment]
Hi do you solve the issue?

**Comment by Akash-Bhiwgade** [comment]
I have not got any solutions yet.

**Comment by dbolya** [comment]
I'm not sure why this happens, but in @Akash-Bhiwgade's comment I see that A has a size of 0:
`... .expand(n, A, B, 2) #Variable[CUDAFloatType]{1,0,19248,2}`
That happens where there are no detections to train on, which borks everything in the loss calculation.

Usually the dataloader should reject samples with no training data, so I'm not sure why this would happen. Have you guys done anything special with the code / maybe marked too many GT as "iscrowd"?

**Comment by Arunavameister** [comment]
Hello dbolya, Thanks for the comment.
I have practically changed nothing in the code, have you ever run this on colab with success?

**Comment by Akash-Bhiwgade** [comment]
I am able to run the code if I set all the iscrowd key to 0.

**Comment by dbolya** [comment]
Note that "iscrowd" means "I'm marking a box over this entire crowd, so you should ignore elements of the crowd because I'm not going to annotate them." It does *not* mean a box annotates a member of a crowd, only use it when you want the detector *to ignore the annotation*.

For most practical purposes, you should tie iscrowd to false (0).

**Comment by Akash-Bhiwgade** [comment]
Hi Dan,

Just a random question on the same. Let's say for instance, we need to use yolact only to give segmentation on priority and give less priority to bounding boxes, what changes should be made and how can we work it to give the best results for segmentation..?

Best regards,
Akash Bhiwgade

**Comment by Arunavameister** [comment]
@Akash-Bhiwgade 
I am still unable to run the code and i keep get the same error, i see that the box_a returns an empty tensor and i cant seem to be able to fix this problem.
Did you do anything else other than changing the iscrowd parameter?

**Comment by jediofgever** [comment]
hi, 
I also stuck here with my custom dataset,
I annotated the images with a tool named VIA, and expoerted annotation file as COCO format. 
There are some images in the json file without any objects in , are they causing this error ?

**Comment by jediofgever** [comment]
@aakashjhawar  iscrowd =0 also does not fix this error for me, have you got it running yet >?

**Comment by sadik1111** [comment]
> Hello,
> 
> I am trying to train a custom dataset(with just a handful images for now) to try out this tool on GOOGLE COLAB, however i am getting the following error:
> !python train.py --config=yolact_base_config --batch_size=1
> 
> Scaling parameters by 0.12 to account for a batch size of 1.
> Per-GPU batch size is less than the recommended limit for batch norm. Disabling batch norm.
> loading annotations into memory...
> Done (t=0.00s)
> creating index...
> index created!
> loading annotations into memory...
> Done (t=0.00s)
> creating index...
> index created!
> Initializing weights...
> Begin training!
> 
> Traceback (most recent call last):
> File "train.py", line 502, in
> train()
> File "train.py", line 305, in train
> losses = net(datum)
> File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py", line 541, in **call**
> result = self.forward(*input, **kwargs)
> File "/usr/local/lib/python3.6/dist-packages/torch/nn/parallel/data_parallel.py", line 150, in forward
> return self.module(*inputs[0], **kwargs[0])
> File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py", line 541, in **call**
> result = self.forward(*input, **kwargs)
> File "train.py", line 144, in forward
> losses = self.criterion(self.net, preds, targets, masks, num_crowds)
> File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py", line 541, in **call**
> result = self.forward(*input, **kwargs)
> File "/content/drive/My Drive/YOLACT/layers/modules/multibox_loss.py", line 124, in forward
> loc_t, conf_t, idx_t, idx, loc_data[idx])
> File "/content/drive/My Drive/YOLACT/layers/box_utils.py", line 182, in match
> overlaps = jaccard(truths, decoded_priors) if not cfg.use_change_matching else change(truths, decoded_priors)
> File "/content/drive/My Drive/YOLACT/layers/box_utils.py", line 73, in jaccard
> inter = intersect(box_a, box_b)
> RuntimeError: strides[cur - 1] == sizes[cur] * strides[cur] INTERNAL ASSERT FAILED at /pytorch/torch/csrc/jit/fuser/executor.cpp:175, please report a bug to PyTorch.
> The above operation failed in interpreter, with the following stack trace:
> 
> Any help will be appreciated!
> Thanks

**Comment by sadik1111** [comment]
> @aakashjhawar iscrowd =0 also does not fix this error for me, have you got it running yet >?
I have changed the "label_map" to correspond to the "class_id" of my coco-format dataset, and there is no more problem. You can check the category_id start at 1, if it is not maybe you can change the label_map to solve the problem

**Comment by XunHang** [comment]
@Arunavameister I have the same question, but it

**Comment by XunHang** [comment]
@Arunavameister 
I have exactly the same error as you, but my error does not appear at the beginning of the program, but only after 20,000 iterations, like 20,320 or 21,100. The following is my error, which is basically the same as yours.

> [  0]   20300 || B: 2.984 | C: 4.802 | M: 4.346 | S: 0.854 | T: 12.986 || ETA: 6:59:27 || timer: 0.650
[  0]   20310 || B: 3.067 | C: 4.785 | M: 4.475 | S: 0.856 | T: 13.183 || ETA: 6:51:26 || timer: 0.823
[  0]   20320 || B: 3.032 | C: 4.841 | M: 4.427 | S: 0.932 | T: 13.231 || ETA: 7:16:22 || timer: 0.658
Traceback (most recent call last):
  File "train.py", line 518, in <module>
    train()
  File "train.py", line 314, in train
    losses = net(datum)
  File "D:\Software\Anaconda\envs\pytorch\lib\site-packages\torch\nn\modules\module.py", line 532, in __call__
    result = self.forward(*input, **kwargs)
  File "D:\Software\Anaconda\envs\pytorch\lib\site-packages\torch\nn\parallel\data_parallel.py", line 150, in forward
    return self.module(*inputs[0], **kwargs[0])
  File "D:\Software\Anaconda\envs\pytorch\lib\site-packages\torch\nn\modules\module.py", line 532, in __call__
    result = self.forward(*input, **kwargs)
  File "train.py", line 148, in forward
    losses = self.criterion(self.net, preds, targets, masks, num_crowds)
  File "D:\Software\Anaconda\envs\pytorch\lib\site-packages\torch\nn\modules\module.py", line 532, in __call__
    result = self.forward(*input, **kwargs)
  File "D:\_Task_\yolact-master\layers\modules\multibox_loss.py", line 124, in forward
    loc_t, conf_t, idx_t, idx, loc_data[idx])
  File "D:\_Task_\yolact-master\layers\box_utils.py", line 182, in match
    overlaps = jaccard(truths, decoded_priors) if not cfg.use_change_matching else change(truths, decoded_priors)
  File "D:\_Task_\yolact-master\layers\box_utils.py", line 73, in jaccard
    inter = intersect(box_a, box_b)
RuntimeError: strides[cur - 1] == sizes[cur] * strides[cur] INTERNAL ASSERT FAILED at ..\torch\csrc\jit\fuser\executor.
cpp:175, please report a bug to PyTorch.
The above operation failed in interpreter.
Traceback (most recent call last):

My pytorch version is 1.4.0, maybe it is a problem of torch?

**Comment by deepakksingh** [comment]
I'm trying to train cityscapes on YOLACT and I'm facing the same issue.
The error is raised at non-regular intervals.
Sometimes at 50th, 120th, or 150th iterations.

**Comment by taketake25** [comment]
I met similar issue for two days...  
However, I can now do this by changing "category_id" number of my datasets.

When I use number of 0 in "category_id", the "INTERNAL ASSERT FAILD" error is appear. 
Changing category_id number to "1" enables my training.

-----
My conditions
* Google Colab using GPU (CUDA 9.2 ?)
* custom dataset (only a category)
* torch==1.4.0,  torchvision==0.5.0
* resnet50-19c8e357.pth


-before-
train/test.json
>      "categories": [{"id": 0, ...}],
>      "annotations":[{"category_id": 0, ...}, ... ]
config.py
>      dataset_base = Config({...
>          'label_map': {0:0},
>      })


-after-
train/test.json
>      "categories": [{"id": 1, ...}],
>      "annotations":[{"category_id": 1, ...}, ... ]
config.py
>      dataset_base = Config({...
>          'label_map': {1:1},
>      })
-----

I hope this helps your training.
Please excuse my poor English.

**Comment by Arunavameister** [comment]
Hello guys just a little update,
After the initial issues, I had stopped using the repo altogether but only yesterday I decided to retry everything again. There was indeed a problem with my annotations which is what was causing the error, the traceback certainly doesn't help to understand where it comes from. To anyone facing similar issues, I strongly recommend rechecking the annotations. 
To everyone who tried to help with your valuable comments, thank you very much.
I will close the issue for now. 
Have a nice day
